imageatm/components/training.py

```diff
@@ -137,18 +155,45 @@ def _fit_model(self):
             save_weights_only=False,
         )
 
+        def _set_patience():
+            """Adjust patience for early stopping and learning rate schedule
```

Comment:
just a thought: is the mean n per class the best indicator or would it be more beneficial to look at the largest/smallest class?

Reply:
With a smallest class approach a single minority class could slow down training significantly with no assurances that model performance benefits from more epochs - so I believe the mean class size is the best we can do without further experimentation.